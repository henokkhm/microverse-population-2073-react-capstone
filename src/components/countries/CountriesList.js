import PropTypes from 'prop-types';

import Country from './Country';
import styles from '../styles/CountriesList.module.css';

function CountriesList({ countryData }) {
  return (
    <div className={styles.countriesListWrapper}>
      <h2 className={styles.countriesListHeader}>{`${countryData.region} Population in 2073`}</h2>
      <ul className={styles.countriesList}>
        {countryData.countries.map((country) => (
          <Country
            key={country.id}
            name={country.name}
            populationIn2073={country.populationIn2073}
          />
        ))}
      </ul>
    </div>
  );
}

const countriesListShape = PropTypes.shape({
  id: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  populationIn2073: PropTypes.number.isRequired,
});

CountriesList.propTypes = {
  countryData: PropTypes.shape({
    region: PropTypes.string.isRequired,
    countries: PropTypes.arrayOf(countriesListShape).isRequired,
  }).isRequired,
};

export default CountriesList;
