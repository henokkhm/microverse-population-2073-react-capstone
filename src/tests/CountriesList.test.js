import { render } from '@testing-library/react';

import CountriesList from '../components/countries/CountriesList';

describe('The CountriesList component', () => {
  it('matches snapshot', () => {
    const { regionId, regionName, countries } = {
      regionId: 'bbf0577d-2cec-44e8-a8c7-a394c2fcca99',
      regionName: 'Asia [mock]',
      countries: [
        {
          id: '1a43497f-1e47-4095-9e8c-72d9141b5d4b',
          name: 'India [mock]',
          populationIn2073: 168404406,
        },
        {
          id: '49943bfc-c183-484e-be45-17468eb39661',
          name: 'China [mock]',
          populationIn2073: 1056989242,
        },
        {
          id: '2920d9fa-57eb-404c-b4fc-647039d43374',
          name: 'Pakistan [mock]',
          populationIn2073: 44714003,
        },
      ],
    };
    const { asFragment } = render(
      <CountriesList
        regionId={regionId}
        regionName={regionName}
        countries={countries}
      />,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
