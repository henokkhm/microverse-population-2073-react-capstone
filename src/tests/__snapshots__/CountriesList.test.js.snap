// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The CountriesList component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="countriesListWrapper"
  >
    <h2
      class="countriesListHeader"
    >
      Asia [mock] Population in 2073
    </h2>
    <ul
      class="countriesList"
      data-region-id="bbf0577d-2cec-44e8-a8c7-a394c2fcca99"
    >
      <li
        class="country"
      >
        <div
          class="countryName"
        >
          India [mock]
        </div>
        <div>
          168,404,406 people
        </div>
        <button
          class="arrowBtn"
          type="button"
        >
          <img
            alt="circled arrow button pointing right"
            class="arrowBtnIcon"
            src="circled-arrow.svg"
          />
        </button>
      </li>
      <li
        class="country"
      >
        <div
          class="countryName"
        >
          China [mock]
        </div>
        <div>
          1,056,989,242 people
        </div>
        <button
          class="arrowBtn"
          type="button"
        >
          <img
            alt="circled arrow button pointing right"
            class="arrowBtnIcon"
            src="circled-arrow.svg"
          />
        </button>
      </li>
      <li
        class="country"
      >
        <div
          class="countryName"
        >
          Pakistan [mock]
        </div>
        <div>
          44,714,003 people
        </div>
        <button
          class="arrowBtn"
          type="button"
        >
          <img
            alt="circled arrow button pointing right"
            class="arrowBtnIcon"
            src="circled-arrow.svg"
          />
        </button>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
